### ISSUE TITLE
Use VueRemark programmaticaly

### ISSUE THREAD

**Issue by RomainLanz** [gridsome/gridsome#984]
Hey all! 👋 

## Summary

I'd like to know if it's possible to use `vue-remark` to inject processed markdown (MDX) into the data layer.

I tried to do the following:
```js
{
  use: '@gridsome/vue-remark',
  options: {
    typeName: 'Partial',
    baseDir: './content/partials',
    plugins: [
      ['gridsome-plugin-remark-shiki', { theme: 'nord', skipInline: true }]
    ]
  }
}
```

The issue in this example is:

- It creates a page automatically with the ID of any "partials"
- The content added in the data layer isn't processed

### Basic example

What I'd like to do is be able to query those in any page and place them whereever I want using any kind of component (since `VueRemarkContent` can only be used in a template).

Given the following query:
```vue
<page-query>
query {
  showCaseTabs: partial(id: "show-case-tabs") {
    content
  }
}
</page-query>
```

I'd like to be able to display the content of this "partial" with the following:
```vue
```

### Motivation

Why are we doing this? What use cases does it support? What is the expected outcome?

**Comment by hjvedvik** [comment]
Yes, it would be very nice to be able to display content like that :)

We have actually been experimenting with GraphQL schema extensions to process Vue in markdown. The idea is that you would use the filesystem source plugin instead, and then process `content` or any other field with the extension. Check out [this readme](https://github.com/gridsome/gridsome/blob/extension-vue-remark/packages/extension-vue/README.md) for a more detailed explanation about how it would be done. What do you think about the schema extension approach?

**Comment by RomainLanz** [comment]
This seems like a good start and I like the idea.

How could it also work with markdown processing?
My goal is to avoid to duplicated the configuration, otherwise I could simply add Remark (& plugins) and use a VueComponent to compute it.

**Comment by hacknug** [comment]
Not sure if I fully understood your needs, but if you have the pre-processed markdown content on your pages, you could use a filter to generate the render-ready HTML like this:

```js
// main.js
import marked from 'marked'
Vue.filter('markdown', (string) => marked(string))
```

```vue
<!-- component.vue -->
<div v-html="$options.filters.markdown(description)" />
```

**Comment by J-Rojas** [comment]
@hacknug your suggestion is very cool, great to know you can renderer markdown so simply!

But I believe this sidesteps Vue-Remark entirely, which would not process any Vue components or imports if they are in the source file. It would be useful to be able to render markdown content that has this structure.
